syntax = 'proto3';

package o2.bookkeeping;

import 'common.proto';

// Services

service RunService {
  rpc Create(RunCreationRequest) returns (Run);
  rpc Update(RunUpdateRequest) returns (Run);
}

// High level messages

message RunCreationRequest {
  string environmentId = 1;
  int32 nDetectors = 2;
  int32 nEpns = 3;
  int32 nFlps = 4;
  int32 runNumber = 5;
  RunType runType = 6;
  bool dd_flp = 7;
  bool dcs = 8;
  bool epn = 9;
  string epnTopology = 10;
  optional string odcTopologyFullName = 11;
  repeated Detector detectors = 12;
}

message RunUpdateRequest {
  int32 runNumber = 1;
  optional int64 timeO2Start = 3;
  optional int64 timeO2End = 4;
  optional int64 timeTrgStart = 5;
  optional int64 timeTrgEnd = 6;
  optional string triggerValue = 7;
  optional string pdpConfigOption = 8;
  optional string pdpTopologyDescriptionLibraryFile = 9;
  optional string tfbDdMode = 10;
  optional string lhcPeriod = 11;
  optional string odcTopologyFullName = 12;
  optional string pdpWorkflowParameters = 13;
  optional string pdpBeamType = 14;
  optional string readoutCfgUri = 15;
}
